### PR TITLE
Hide `Adversary` instead of `Adversary.Perturber`

### PR DESCRIPTION
**Pull request by mzweilin**
# What does this PR do?

Hide the `Adversary` modules instead of `Adversary.Perturber`, so that we can make use of all `Trainer` features in `Adversary`.

We may configure not to hide `Adversary` for computing universal perturbations.

Depends on #144

## Type of change

Please check all relevant options.

- [ ] Improvement (non-breaking)
- [ ] Bug fix (non-breaking)
- [x] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Testing

Please describe the tests that you ran to verify your changes. Consider listing any relevant details of your test configuration.

- [x] `pytest`
- [x] `python -m mart experiment=CIFAR10_CNN_Adv trainer=gpu trainer.precision=16` reports 70.23%

## Before submitting

- [x] The title is **self-explanatory** and the description **concisely** explains the PR
- [x] My **PR does only one thing**, instead of bundling different changes together
- [x] I list all the **breaking changes** introduced by this pull request
- [x] I have commented my code
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have run pre-commit hooks with `pre-commit run -a` command without errors

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by mzweilin** [comment]
I tried to run an experiment, but I still didn't see gradient logs on Tensorboard.

```
CUDA_VISIBLE_DEVICES=0 \
python -m mart \
experiment=CIFAR10_CNN_Adv \
fit=false \
trainer=gpu \
+trainer.limit_test_batches=1 \
+callbacks@model.modules.input_adv_test.callbacks=gradient_monitor \
+model.modules.input_adv_test.callbacks.gradient_monitor.frequency=1
```

`pl_module.log_dict(norms)` does receive norms dict, but I couldn't find them on Tensorboard. @dxoigmn

**Comment by dxoigmn** [comment]
I also doubt there a logger associated with the adversary's trainer.

File: mart/attack/adversary.py [hunk 1]
```diff
@@ -76,7 +75,7 @@ def __init__(
             self._attacker = partial(
                 pl.Trainer,
                 num_sanity_val_steps=0,
-                logger=False,
+                logger=list(kwargs.pop("logger", {}).values()),
```
Comment (dxoigmn) [inline mart/attack/adversary.py:78]: I think this should go into a different PR, in part, because, I think it may be better to inherit the logger from the other trainer.

Comment (dxoigmn) [reply mart/attack/adversary.py:78]: Looks like there isn't an easy way to get the base logger.

**Comment by mzweilin** [comment]
Close in favor of #160
